lab_1/main.py

```diff
@@ -30,7 +59,21 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+
+    if not isinstance(tokens, list):
+        return None
+
+    freq_dict = {}
+    for word in tokens:
+        if word not in freq_dict:
+            freq_dict[word] = 1
+        else:
+            freq_dict[word] += 1
+    return freq_dict
+
+tokens_en = remove_stop_words(tokens_en, stop_words_en)
+tokens_de = remove_stop_words(tokens_de, stop_words_de)
+tokens_unknown = remove_stop_words(tokens_unknown, stop_words_unknown)
```

Comment:
The `main.py` file should not contain function calls - you should check your implementation using the `start.py` file.